### DISCUSSION TITLE
NFT Minting Contract (TokenV2 & PropertyMap)

### DISCUSSION THREAD

**Discussion by unbound26z** [aptos-labs/aptos-developer-discussions#155] [Questions] [answered]
Hi, is there any known contract or standard that is used as a candy machine to mint NFTs? I want to find a way to mint 10k NFTs in a safe and secure way. I have around 16 different traits, and each of them has 5-10 values. 

Questions that have been popping up are:
1. What's the best way to store the "candies" in the machine? Make an add_1_candy or add_10_candies and spam that with a script, once i have jsons ready off chain? Candies would be objects that have all needed properties and info for the nft
2. Later, when minting, will people be able to simulate and somehow target some rare traits if I have them all sitting in a candy machine, and I use pseudorandomness?
3. Are property maps still used in token v2? Should i on the mint function add duplication, so that all of the candy info from the candy machine we randomly pull get copied over to the property map for wallets to showcase?

I assume I'd have even more questions if I thought even more deeply, but if someone could answer this, it would push me in the right direction.

### What have you tried or looked at?  Or how can we reproduce the error?

Not relevant.

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

TypeScript SDK

### Describe your environment or tooling in detail

Not relevant.

**Comment by gregnazario** [comment]
@JohnChangUK

**Comment by JohnChangUK** [comment, marked as answer]
We are building a minting standard as part of [AIP 72](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-72.md), which will provide a range of modules and examples to create collections and tokens. The [token-minter repo](https://github.com/aptos-labs/token-minter) contains the move components for collection and token creation; to use under `/token-minter`. The other repos provide examples on how to use this repo, along with other features such as composability and extending the tokens. 

For randomness, we have true randomness coming soon! So for now, you can use psuedorandomness, and switch to the `randomness` module once that is released on mainnet.

We are working to put together something similar to what you have asked for (creating NFTs with randomized traits, trait uploading etc). Will provide more details once this is being progressed.
